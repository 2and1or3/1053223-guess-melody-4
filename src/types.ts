enum GameType {
  ARTIST = `artist`,
  GENRE = `genre`,
}

interface GenreAnswer {
  genre: string;
  src: string;
}

interface ArtistAnswer {
  artist: string;
  pictureSrc: string;
}

interface GenreQuestion {
  type: GameType.GENRE;
  genre: string;
  answers: Array<GenreAnswer>;
}

interface ArtistQuestion {
  type: GameType.ARTIST;
  artist: string;
  trackSrc: string;
  answers: Array<ArtistAnswer>;
}

export { GenreAnswer, ArtistAnswer, GenreQuestion, ArtistQuestion, GameType };
