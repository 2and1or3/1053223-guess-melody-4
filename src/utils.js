import {GameType} from './types.ts';

const mockAudioNode = (element) => {
  if (element.type === `audio`) {
    return element;
  }

  return null;
};

const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const adapterForGenreType = (question) => question;

const adapterForArtistType = (question) => ({
  type: question.type,
  artist: question.song.artist,
  trackSrc: question.song.src,
  answers: question.answers.map((answer) => ({
    artist: answer.artist,
    pictureSrc: answer.picture,
  }))
});

const adapterToLocalQuestions = (questions) => {

  const localQuestions = questions.map((question) => {
    switch (question.type) {
      case (GameType.ARTIST):
        return adapterForArtistType(question);
      case (GameType.GENRE):
        return adapterForGenreType(question);

      default:
        return question;
    }
  });

  return localQuestions;
};

export {mockAudioNode, adapterToLocalQuestions, extend};
