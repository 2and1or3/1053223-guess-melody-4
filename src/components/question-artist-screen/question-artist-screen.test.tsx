import * as React from "react";
import * as renderer from "react-test-renderer";

import QuestionArtistScreen from './question-artist-screen';
import {ArtistQuestion, GameType} from '../../types';
import {noop} from '../../utils';

const question: ArtistQuestion = {
  type: GameType.ARTIST,
  artist: `true-person`,
  trackSrc: `src-for-track`,
  answers: [
    {
      artist: `true-person`,
      pictureSrc: `http://placehold.it/134x134`,
    },
    {
      artist: `person-2`,
      pictureSrc: `http://placehold.it/134x134`,
    },
    {
      artist: `person-3`,
      pictureSrc: `http://placehold.it/134x134`,
    },
  ]
};

it(`Render QuestionArtistScreen component`, () => {
  const tree = renderer
    .create(<QuestionArtistScreen
      question = {question}
      onAnswer = {noop}
      renderPlayer = {noop}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
