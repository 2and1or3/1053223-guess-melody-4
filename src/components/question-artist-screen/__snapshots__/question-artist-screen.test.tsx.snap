// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render QuestionArtistScreen component 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="true-person0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="true-person"
      />
      <label
        className="artist__name"
        htmlFor="true-person0"
      >
        <img
          alt="true-person"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        true-person
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="person-21"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="person-2"
      />
      <label
        className="artist__name"
        htmlFor="person-21"
      >
        <img
          alt="person-2"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        person-2
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="person-32"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="person-3"
      />
      <label
        className="artist__name"
        htmlFor="person-32"
      >
        <img
          alt="person-3"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        person-3
      </label>
    </div>
  </form>
</section>
`;
